### PR TITLE
Add pid file creation for okteto up command.

### LINKED ISSUE #691: generate a pid file
**Is your feature request related to a problem? Please describe.**
okteto should generate a PID file on `up`, so that other tools can monitor it's state and existence

### PR DESCRIPTION
**Pull request by adhaamehab**
Signed-off-by: adhaamehab <adhaamehab.me@gmail.com>

Fixes #691 

## Proposed changes
- Okteto Up generates a `namespace-deployment.pid` file at `.okteto/` folder defined in the `$OKTETOHOME`
- The pid file gets deleted once `okteto up` finishes
- If the namespace isn't defined in `okteto.yml` file or `--namespace` flag. we use the namespace provided by the k8s client

### CODE REVIEW

File: cmd/up.go [hunk 1]
```diff
@@ -897,3 +914,30 @@ func printDisplayContext(dev *model.Dev) {
 	}
 	fmt.Println()
 }
+
+// createPIDFile creates a PID file to track Up state and existence
+func createPIDFile(ns, dp string) error {
+
+	fileName := fmt.Sprintf("%s-%s.pid", ns, dp)
+	filePath := filepath.Join(config.GetOktetoHome(), fileName)
```
Comment (rberrelleza) [inline cmd/up.go:922]: I think is better if the file is called `okteto.pid` and you have it inside `config.GetDeploymentHome`, that's the folder that's meant to keep the files specific to one `okteto up`

File: cmd/up.go [hunk 2]
```diff
@@ -897,3 +914,30 @@ func printDisplayContext(dev *model.Dev) {
...
+	fmt.Println(filePath)
```
Comment (rberrelleza) [inline cmd/up.go:923]: log?

File: cmd/up.go [hunk 3]
```diff
@@ -897,3 +914,30 @@ func printDisplayContext(dev *model.Dev) {
 	}
 	fmt.Println()
 }
+
+// createPIDFile creates a PID file to track Up state and existence
+func createPIDFile(ns, dp string) error {
```
Comment (rberrelleza) [inline cmd/up.go:919]: could you add  unit test for this?

File: cmd/up.go (hunk 2 above)
Comment (adhaamehab) [reply cmd/up.go:923]: yes, sorry about that!a

File: cmd/up.go (hunk 3 above)
Comment (adhaamehab) [reply cmd/up.go:919]: Sure, will add it right away!

File: cmd/up.go [hunk 4]
```diff
@@ -162,6 +165,20 @@ func Up() *cobra.Command {
 
 //RunUp starts the up sequence
 func RunUp(dev *model.Dev, autoDeploy, build, forcePull, resetSyncthing bool) error {
+
+	if dev.Namespace == "" {
+		_, _, namespace, err := k8Client.GetLocal()
+		if err != nil {
+			return err
+		}
+		dev.Namespace = namespace
```
Comment (pchico83) [inline cmd/up.go:174]: The namespace value might come from the namespace flag too.
`dev.Namespace` is initialized at this point:
    https://github.com/okteto/okteto/blob/67e1c712d3251d2397d93c1e11349c43392eed42/cmd/up.go#L232
The `RunUp` logic is a bit messy, we need to refactor it. But you could bring the `dev.Namespace` initialization logic here if needed

Comment (adhaamehab) [reply cmd/up.go:174]: Sorry, @pchico83  I'm a little bit confused because the referenced line is older than the current. Can you explain more?

Comment (pchico83) [reply cmd/up.go:174]: Sure. In the master branch, `dev.Namespace` is initialized from lines 220-230 in `cmd/up.go`.
I think you can move that logic here to have a single place where where `k8Client.GetLocal()` is called:
Screenshot 2020-05-04 at 21 45 24

Comment (adhaamehab) [reply cmd/up.go:174]: Got it, thanks.
Will do so 👍🏻

**Comment by adhaamehab** [comment]
@rberrelleza @pchico83 
Can you review the latest changes whenever you have time?

File: cmd/up.go [hunk 5]
```diff
@@ -897,3 +908,25 @@ func printDisplayContext(dev *model.Dev) {
...
+	defer file.Close()
+	if _, err := file.WriteString(strconv.Itoa(os.Getpid())); err != nil {
+		return fmt.Errorf("Unable to write to PID file at %s", filePath)
+	}
+	return nil
+}
+
+// cleanPIDFile deletes PID file after Up finishes
+func cleanPIDFile(ns, dpName string) {
+	filePath := filepath.Join(config.GetOktetoHome(), "okteto.pid")
```
Comment (rberrelleza) [inline cmd/up.go:928]: I think this needs to be `config.GetDeploymentHome` too.

File: cmd/up.go [hunk 6]
```diff
@@ -897,3 +908,25 @@ func printDisplayContext(dev *model.Dev) {
...
+	defer file.Close()
+	if _, err := file.WriteString(strconv.Itoa(os.Getpid())); err != nil {
+		return fmt.Errorf("Unable to write to PID file at %s", filePath)
+	}
+	return nil
+}
+
+// cleanPIDFile deletes PID file after Up finishes
+func cleanPIDFile(ns, dpName string) {
+	filePath := filepath.Join(config.GetDeploymentHome(), "okteto.pid")
```
Comment (pchico83) [inline cmd/up.go:928]: @adhaamehab tests are failing. I think you are missing the args here:
```
GetDeploymentHome(ns, dpName)
```

**Comment by rberrelleza** [comment]
Thanks for adding the test, looks great!

**Pull request merged**